### PR TITLE
Add xgboost integration

### LINKED ISSUE #453: Xgboost integration
Add Xgboost integration

### PR DESCRIPTION
**Pull request by Khazhak**
Closes #453.

### CODE REVIEW

File: aim/sdk/adapters/xgboost.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+from xgboost.callback import TrainingCallback, CallbackContainer
```
Comment (gorarakelyan) [inline aim/sdk/adapters/xgboost.py:1]: lets implement an interface like PL, huggingface integrations to not have direct dependency from xgboost

Comment (gorarakelyan) [reply aim/sdk/adapters/xgboost.py:1]: e.g.
```
class AimCallback(object):
    @staticmethod
    def __new__(cls, *args, **kwargs):
        xgboost_logger_inst = cls.__get_logger_cls()
        return xgboost_logger_inst(*args, **kwargs)

    @classmethod
    def __get_logger_cls(cls):
        ...
        from xgboost.callback import TrainingCallback
        class _XGBoostLogger(TrainigCallback):
            ...
```

File: aim/sdk/adapters/xgboost.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from aim.sdk.session.session import Session
+from typing import Optional
+
+class AimCallback(TrainingCallback):
+    def __init__(self, repo: Optional[str] = None,
+                 experiment: Optional[str] = None):
```
Comment (gorarakelyan) [inline aim/sdk/adapters/xgboost.py:7]: Argument `system_tracking_interval: Optional[int] = DEFAULT_SYSTEM_TRACKING_INT` is missing

File: aim/sdk/adapters/xgboost.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        self.initialized = True
+        return model
+
+    def after_iteration(self, model, epoch: int,
+                        evals_log: CallbackContainer.EvalsLog) -> bool:
+        if not evals_log:
+            return False
+
+        for data, metric in evals_log.items():
+            for metric_name, log in metric.items():
```
Comment (gorarakelyan) [inline aim/sdk/adapters/xgboost.py:27]: Do metric names have prefixes that define the training phase? e.g. `train_`, `val_`. If yes, then lets automatically set corresponding `subset` context.

File: aim/sdk/adapters/xgboost.py (hunk 2 above)
Comment (gorarakelyan) [reply aim/sdk/adapters/xgboost.py:7]: + `flush_frequency: int = DEFAULT_FLUSH_FREQUENCY`

File: aim/sdk/adapters/xgboost.py (hunk 3 above)
Comment (Khazhak) [reply aim/sdk/adapters/xgboost.py]: there were no such prefixes, no `train_begin` or `val_begin` functions as well. I couldn't find a way to get the current phase.

**Comment by gorarakelyan** [comment]
Callback implementation looks great. Can we add training code example as well?
